klayout/drc/run_drc.py

```diff
@@ -0,0 +1,709 @@
+        exit()
+
+    if arguments["--verbose"]:
+        switches["verbose"] = "true"
+    else:
+        switches["verbose"] = "false"
+
+    if arguments["--no_feol"]:
+        switches["feol"] = "false"
+    else:
+        switches["feol"] = "true"
+
+    if arguments["--no_beol"]:
+        switches["beol"] = "false"
+    else:
+        switches["beol"] = "true"
+
+    if arguments["--no_offgrid"]:
+        switches["offgrid"] = "false"
+    else:
+        switches["offgrid"] = "true"
+
+    if arguments["--connectivity"]:
+        switches["conn_drc"] = "true"
+    else:
+        switches["conn_drc"] = "false"
+
+    if arguments["--density"]:
+        switches["density"] = "true"
+    else:
+        switches["density"] = "false"
+
+    switches["topcell"] = get_run_top_cell_name(arguments, layout_path)
+    switches["input"] = layout_path
+
+    return switches
+
+
+def check_klayout_version():
+    """
+    check_klayout_version checks klayout version and makes sure it would work with the DRC.
+    """
+    # ======= Checking Klayout version =======
+    klayout_v_ = os.popen("klayout -b -v").read()
+    klayout_v_ = klayout_v_.split("\n")[0]
+    klayout_v_list = []
+
+    if klayout_v_ == "":
+        logging.error("Klayout is not found. Please make sure klayout is installed.")
+        exit(1)
+    else:
+        klayout_v_list = [int(v) for v in klayout_v_.split(" ")[-1].split(".")]
+
+    logging.info(f"Your Klayout version is: {klayout_v_}")
+
+    if len(klayout_v_list) < 1 or len(klayout_v_list) > 3:
+        logging.error("Was not able to get klayout version properly.")
+        exit(1)
+    elif len(klayout_v_list) == 2:
+        if klayout_v_list[1] < 28:
+            logging.warning("Prerequisites at a minimum: KLayout 0.28.0")
+            logging.error(
+                "Using this klayout version has not been assesed in this development. Limits are unknown"
+            )
+            exit(1)
+    elif len(klayout_v_list) == 3:
+        if klayout_v_list[1] < 28 :
+            logging.warning("Prerequisites at a minimum: KLayout 0.28.0")
+            logging.error(
+                "Using this klayout version has not been assesed in this development. Limits are unknown"
+            )
+            exit(1)
+
+
+def check_layout_path(layout_path):
+    """
+    check_layout_type checks if the layout provided is GDS or OAS. Otherwise, kill the process. We only support GDS or OAS now.
+
+    Parameters
+    ----------
+    layout_path : string
+        string that represent the path of the layout.
+
+    Returns
+    -------
+    string
+        string that represent full absolute layout path.
+    """
+
+    if not os.path.isfile(layout_path):
+        logging.error("## GDS file path provided doesn't exist or not a file.")
+        exit(1)
+
+    if ".gds" not in layout_path and ".oas" not in layout_path:
+        logging.error("## Layout is not in GDSII or OASIS format. Please use gds format.")
```

Comment:
we should make sure we print the actual path as part of the error message.

Reply:
done